components/render/render.h

```diff
@@ -36,6 +36,34 @@ namespace Level
 
 namespace Render
 {
+    // Difference between two points
+    struct Vector
+    {
+      int32_t x;
+      int32_t y;
+
+      Vector operator* (const double &scalar) const
+      {
+        return {static_cast<int32_t> (x * scalar), static_cast<int32_t> (y * scalar)};
+      }
+    };
+
+    // Point in pixels
+    struct Point
+    {
+      int32_t x;
+      int32_t y;
+      Point operator+ (const Vector &v) const { return {x + v.x, y + v.y}; }
+      Point operator- (const Vector &v) const { return {x - v.x, y - v.y}; }
+      Vector operator- (const Point &other) const { return {x - other.x, y - other.y}; }
+    };
+
+    // Tile mesasured in indexes on tile grid
+    struct Tile
```

Comment:
Generally I think more types are better because of type-safety (e.g. not allowing to pass Tile where argument is Point for example). I'm fine with combining Point and Vector and moving them to Util, I do think that Render::Tile should stay though because later I added possibility to disntiguish left and right halves of a Tile which are distinguished in original game for targetting by mouse.